### PR TITLE
SIDE-1686: adding go loop documentation

### PR DESCRIPTION
**Pull request by RusGraf**

### CODE REVIEW

File: ldk/go/readme.md [hunk 1]
```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
...
+		return err
+	}
+
+	if err := sidekick.UI().ListenSearchbar(l.ctx, handler); err != nil {
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
```
Comment (BrettMugglinOlive) [inline ldk/go/readme.md:61]: Beginning each sentence here with "Loops" feels a little...samey? It just doesn't flow very well.

It doesn't have to be exactly this, but something like

"Loops do not need to emit whispers in a 1:1 relationship with events, and may not use events at all. They could only use some events, keep a history of events, or only emit whispers when several conditions are met."

**Review by BrettMugglinOlive — APPROVED** [review]

File: ldk/go/readme.md [hunk 2]
```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
...
+Writing a loop boils down to writing an implementation for the `Loop` interface and serving it via `ldk.ServeLoopPlugin`.
+
+```go
+type Loop interface {
+	LoopStart(Sidekick) error
+	LoopStop() error
+}
+
+func main() {
+	...
```
Comment (richseviora) [inline ldk/go/readme.md:30]: in place of the ellipsis, could you show them how we instantiate the logger, and maybe a placeholder for the Loop struct instantiation

File: ldk/go/readme.md [hunk 3]
```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
...
-Example loops are available in the examples folder.
+This LDK can be used to write loops for Olive Helps. An example of the loop implementation is available [here](docs/loops.md).
+
+
+#### Interface
+
+Writing a loop boils down to writing an implementation for the `Loop` interface and serving it via `ldk.ServeLoopPlugin`.
+
+```go
+type Loop interface {
```
Comment (richseviora) [inline ldk/go/readme.md:24]: This reads as a part of the code below, and I don't _think_ it should be? Maybe break it from the `main` function below as a separate codeblock, so it'd be like:
```go
type Loop interface {
  LoopStart(ldk.Sidekick) error
  LoopStop() error
}
```

```go
import $LDK_MODULE$

func main() {
  # Instantiate Loop
  # Instantiate Logger
  ldk.ServeLoopPlugin(logger, loop)
}
```

File: ldk/go/readme.md [hunk 4]
```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
...
+
+	if err := sidekick.UI().ListenSearchbar(l.ctx, handler); err != nil {
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
+
+#### List of possible Sensors to subscribe to
```
Comment (richseviora) [inline ldk/go/readme.md:63]: Now that I think about it, and taking to account the developer hub aptitude list we have available, maybe we should link to that instead...

File: ldk/go/docs/loops.md [hunk 5]
```diff
@@ -1,51 +1,108 @@
...
+type Loop struct {
+	ctx              context.Context
+	cancel           context.CancelFunc
+	sidekick         ldk.Sidekick
+	logger           *ldk.Logger
+}
 
-**LoopStop** - The Loop should stop operating when this is called.
+// LoopStart will be called by ldk when loop starts
+func (loop *Loop) LoopStart(sidekick ldk.Sidekick) error {
```
Comment (richseviora) [inline ldk/go/docs/loops.md:33]: Go's [code style](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names) prefers that receiver variable names are one or two character

**Review by richseviora — CHANGES_REQUESTED** [review]
Some changes :)

File: ldk/go/readme.md (hunk 1 above)
Comment (RusGraf) [reply ldk/go/readme.md]: done

File: ldk/go/readme.md (hunk 2 above)
Comment (RusGraf) [reply ldk/go/readme.md]: done

File: ldk/go/readme.md (hunk 3 above)
Comment (RusGraf) [reply ldk/go/readme.md:24]: Separated.

File: ldk/go/readme.md (hunk 4 above)
Comment (RusGraf) [reply ldk/go/readme.md:73]: Not a bad idea. Do you have a link to where would we like to point them?

File: ldk/go/docs/loops.md (hunk 5 above)
Comment (RusGraf) [reply ldk/go/docs/loops.md]: Understood.

File: ldk/go/readme.md (hunk 4 above)
Comment (richseviora) [reply ldk/go/readme.md:73]: Hmm... not a great link right now - but I like having the interface available instead.

Oops! one thing though, we need to replace references to "Sensors" with "Aptitudes" in the documentation, we don't call them Sensors any more so we should avoid them in the documents.

Comment (RusGraf) [reply ldk/go/readme.md:73]: Ahh, I did that and then changed it back based on the response in the slack. I'll change it again.

Comment (RusGraf) [reply ldk/go/readme.md]: Updated

**Review by richseviora — APPROVED** [review]

**Pull request merged**